### PR TITLE
Add Code engine secret resource

### LINKED ISSUE #4431: Code Engine: Ability to manage secrets and service bindings
Hi.

I know that the CE resources aren't released yet, but while I've been awaiting the release I've built the provider locally and started playing around with it while referring to the markdown docs in the repo. 

As far I can tell, there is nothing [here](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/website/docs/r/code_engine_config_map.html.markdown) that lets me create a secret instead of a normal config map, even though `run_env_variables`'s `type` lets me specify `secret_full_reference` and the like. I figured it might be possible that the secrets were stored in another service(like the Secrets Manager) and I would need to create them that way, but from looking in my SM instance, there are no references to any of the secrets I've created via Code Engine. Am I missing something?

Another thing I'd also like is the ability to create service bindings. None of the new code engine resources that were added seem to have anything to do with service bindings, but from looking in the provider docs, there are a couple of possibilities, like `ibm_service_key`, `ibm_iam_service_id`/`ibm_iam_service_api_key` and possibly `ibm_container_bind_service`, but I'm not sure which would be the right ones and what combination allows me to create what amounts to a Code Engine service binding.

### PR DESCRIPTION
**Pull request by michael-magrian**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #4431

Output from acceptance testing:

```
$ make testacc                                                 
TF_ACC=1 go test -v $(go list ./... |grep -v 'vendor') 
?       github.com/IBM-Cloud/terraform-provider-ibm     [no test files]
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest [no test files]
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex    [no test files]
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider        [no test files]
=== RUN   TestString
--- PASS: TestString (0.00s)
=== RUN   TestString_positiveIndex
--- PASS: TestString_positiveIndex (0.00s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns   (cached)
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate        [no test files]
?       github.com/IBM-Cloud/terraform-provider-ibm/version     [no test files]
=== RUN   TestAccIbmCodeEngineProjectDataSourceBasic
--- PASS: TestAccIbmCodeEngineProjectDataSourceBasic (8.27s)
=== RUN   TestAccIbmCodeEngineAppBasic
--- PASS: TestAccIbmCodeEngineAppBasic (146.47s)
=== RUN   TestAccIbmCodeEngineAppExtended
--- PASS: TestAccIbmCodeEngineAppExtended (147.04s)
=== RUN   TestAccIbmCodeEngineBuildBasic
--- PASS: TestAccIbmCodeEngineBuildBasic (11.60s)
=== RUN   TestAccIbmCodeEngineConfigMapBasic
--- PASS: TestAccIbmCodeEngineConfigMapBasic (24.04s)
=== RUN   TestAccIbmCodeEngineJobBasic
--- PASS: TestAccIbmCodeEngineJobBasic (21.34s)
=== RUN   TestAccIbmCodeEngineJobExtended
--- PASS: TestAccIbmCodeEngineJobExtended (22.90s)
=== RUN   TestAccIbmCodeEngineSecretGeneric
--- PASS: TestAccIbmCodeEngineSecretGeneric (23.64s)
=== RUN   TestAccIbmCodeEngineSecretBasicAuth
--- PASS: TestAccIbmCodeEngineSecretBasicAuth (23.74s)
=== RUN   TestAccIbmCodeEngineSecretRegistry
--- PASS: TestAccIbmCodeEngineSecretRegistry (23.20s)
=== RUN   TestAccIbmCodeEngineSecretSSHAuth
--- PASS: TestAccIbmCodeEngineSecretSSHAuth (22.90s)
=== RUN   TestAccIbmCodeEngineSecretTls
--- PASS: TestAccIbmCodeEngineSecretTls (21.93s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/codeengine      497.660s
...
```

### CODE REVIEW

File: ibm/service/codeengine/resource_ibm_code_engine_app.go [hunk 1]
```diff
@@ -752,6 +751,10 @@ func resourceIbmCodeEngineAppUpdate(context context.Context, d *schema.ResourceD
 	hasChange := false
 
 	patchVals := &codeenginev2.AppPatch{}
+	if d.HasChange("project_id") {
```
Comment (hkantare) [inline ibm/service/codeengine/resource_ibm_code_engine_app.go:754]: We already defined ForceNew: true on "project_id" we don't need to add this condition

File: ibm/service/codeengine/resource_ibm_code_engine_secret.go [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+		return diag.FromErr(err)
+	}
+
+	replaceSecretOptions.SetProjectID(parts[0])
+	replaceSecretOptions.SetName(parts[1])
+	replaceSecretOptions.SetFormat(d.Get("format").(string))
+
+	hasChange := false
+
+	if d.HasChange("project_id") {
```
Comment (hkantare) [inline ibm/service/codeengine/resource_ibm_code_engine_secret.go:258]: Same as above  ForceNew controls this behaviour

File: ibm/service/codeengine/resource_ibm_code_engine_app.go (hunk 1 above)
Comment (michael-magrian) [reply ibm/service/codeengine/resource_ibm_code_engine_app.go]: Removed, thanks for that.

File: ibm/service/codeengine/resource_ibm_code_engine_secret.go (hunk 2 above)
Comment (michael-magrian) [reply ibm/service/codeengine/resource_ibm_code_engine_secret.go]: Removed

**Comment by michael-magrian** [comment]
I addressed your comments, can you please take another look @hkantare

**Pull request merged**